huaweicloud/services/coc/resource_huaweicloud_coc_document.go

```diff
@@ -0,0 +1,330 @@
+		d.Set("description", utils.PathSearch("data.description", document, nil)),
+		d.Set("create_time", utils.PathSearch("data.create_time", document, nil)),
+		d.Set("update_time", utils.PathSearch("data.update_time", document, nil)),
+		d.Set("version", utils.PathSearch("data.version", document, nil)),
+		d.Set("creator", utils.PathSearch("data.creator", document, nil)),
+		d.Set("modifier", utils.PathSearch("data.modifier", document, nil)),
+		d.Set("versions", flattenDocumentVersions(
+			utils.PathSearch("data.versions", document, nil))),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenDocumentVersions(rawParams interface{}) []map[string]interface{} {
+	if paramsList, ok := rawParams.([]interface{}); ok {
+		if len(paramsList) < 1 {
+			return nil
+		}
+		configurations := make([]map[string]interface{}, len(paramsList))
+		for i, params := range paramsList {
+			raw := params.(map[string]interface{})
+			configurations[i] = map[string]interface{}{
+				"version":      utils.PathSearch("version", raw, nil),
+				"version_uuid": utils.PathSearch("version_uuid", raw, nil),
+				"create_time":  utils.PathSearch("create_time", raw, nil),
+			}
+		}
+		return configurations
+	}
+
+	return nil
+}
+
+func GetDocument(client *golangsdk.ServiceClient, documentID string) (interface{}, error) {
+	getHttpUrl := "v1/documents/{document_id}"
+	getPath := client.Endpoint + getHttpUrl
+	getPath = strings.ReplaceAll(getPath, "{document_id}", documentID)
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+	}
+
+	getResp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return nil, err
+	}
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return nil, fmt.Errorf("error flattening document: %s", err)
+	}
+
+	return getRespBody, nil
+}
+
+func resourceDocumentUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+
+	client, err := cfg.NewServiceClient("coc", cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating COC client: %s", err)
+	}
+
+	updateHttpUrl := "v1/documents/{document_id}"
+	updatePath := client.Endpoint + updateHttpUrl
+	updatePath = strings.ReplaceAll(updatePath, "{document_id}", d.Id())
+	updateOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildUpdateDocumentBodyParams(d)),
+	}
+
+	_, err = client.Request("PUT", updatePath, &updateOpt)
+	if err != nil {
+		return diag.Errorf("error updating document: %s", err)
+	}
+
+	return resourceDocumentRead(ctx, d, meta)
+}
+
+func buildUpdateDocumentBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        d.Get("name"),
+		"content":     d.Get("content"),
+		"description": utils.ValueIgnoreEmpty(d.Get("description")),
```

Comment:
should be able to set description to empty
```suggestion
		"description": d.Get("description"),
```